fr.esrf.tango.generator.xtend/src/fr/esrf/tango/pogo/generator/python/PythonDeviceHL.xtend

```diff
@@ -208,15 +208,35 @@ __all__ = ["«cls.name»", "main"]
     //    Constructors
     //====================================================
     def pythonConstructors(PogoDeviceClass cls)  '''
-«IF true»«cls.constructorHL»
+«IF true»    def init_device(self):
+        """ Initializes the attributes and properties of the «cls.name»."""
+«cls.constructorHL»
         «cls.setEventCriteria»
-        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»«cls.protectedAreaHL("init_device")»«ENDIF»
+        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
+        «cls.openProtectedAreaHL("init_device")»
+        «cls.closeProtectedAreaHL("init_device")»
+        «ENDIF»
 
     def always_executed_hook(self):
-        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»«cls.protectedAreaHL("always_executed_hook", "pass", false)»«ELSE»pass«ENDIF»
+        """ Method always executed before any TANGO command is executed."""
+        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»
+        «cls.openProtectedAreaHL("always_executed_hook")»
+        «cls.closeProtectedAreaHL("always_executed_hook")»
+        «ELSE»
+        pass
+        «ENDIF»
 
     def delete_device(self):
-        «IF cls.description.filestogenerate.toLowerCase.contains("protected regions")»«cls.protectedAreaHL("delete_device", "pass", false)»«ELSE»pass«ENDIF»
+        """ Hook to delete resources allocated in init_device.
+        This method allows for any memory or other resources allocated in the
```

Comment:
Need a blank line between first line of docstring (summary) and the rest.

Reply:
Yes will add it.

Reply:
Done. Link to the commit
https://github.com/ska-telescope/pogo/commit/60b7ef2651fc88fc6d518d15e00c8fa216d384a4